### ISSUE TITLE
hailctl dataproc submit does not support `--region` flag

### ISSUE THREAD

**Issue by agraubert** [hail-is/hail#8078]
Using hail 0.2.32, `hailctl dataproc submit` results in the following error:

```
hailctl dataproc submit hail-test {python file} -- {arguments to script}
Submitting to cluster 'hail-test'...
gcloud command:
gcloud dataproc jobs submit pyspark {python file} \
    --cluster=hail-test \
    --files= \
    --py-files=/var/folders/7y/hvrzyxts3xg74r3m2jbq0kc0zt3g3z/T/pyscripts_srh2ze4a.zip \
    --properties= \
    -- \
    {arguments to script}
ERROR: (gcloud.dataproc.jobs.submit.pyspark) The required property [region] is not currently set.
It can be set on a per-command basis by re-running your command with the [--region] flag.

You may set it for your current workspace by running:

  $ gcloud config set dataproc/region VALUE

or it can be set temporarily by the environment variable [CLOUDSDK_DATAPROC_REGION]
Traceback (most recent call last):
  File "/Users/aarong/Documents/gtex-wgs/.devenv/bin/hailctl", line 8, in <module>
    sys.exit(main())
  File "/Users/aarong/Documents/gtex-wgs/.devenv/lib/python3.7/site-packages/hailtop/hailctl/__main__.py", line 94, in main
    cli.main(args)
  File "/Users/aarong/Documents/gtex-wgs/.devenv/lib/python3.7/site-packages/hailtop/hailctl/dataproc/cli.py", line 107, in main
    jmp[args.module].main(args, pass_through_args)
  File "/Users/aarong/Documents/gtex-wgs/.devenv/lib/python3.7/site-packages/hailtop/hailctl/dataproc/submit.py", line 78, in main
    check_call(cmd)
  File "/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py", line 363, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError
``` 

However, adding `--region us-central1` to any location in the argument string to hailctl dataproc submit results in the argument being picked up as an input to the script, not to the underlying gcloud command

**Comment by tpoterba** [comment]
Yeah, this was broken in a recent update to the gcloud libraries. For now, following the command to do:
```
gcloud config set dataproc/region VALUE
```
will generate the best user experience. We'll fix this for the next version though.

**Comment by johnc1231** [comment]
Will be fixed as a side effect of #8281
